### PR TITLE
Improve samples/sieve.d

### PR DESCRIPTION
**Pull request by majiang**
partial remedy for Issue 11710

### CODE REVIEW

File: samples/sieve.d [hunk 1]
```diff
@@ -1,41 +1,32 @@
-
 /* Eratosthenes Sieve prime number calculation. */
 
+import std.conv;
```
Comment (majiang) [inline samples/sieve.d]: see L8-10

**Comment by WalterBright** [comment]
https://issues.dlang.org/show_bug.cgi?id=11710

File: samples/sieve.d [hunk 2]
```diff
@@ -1,41 +1,32 @@
...
     {
-        count   = 0;
-        flags[] = true;
+        if (!flags[i])
+            continue;
+        auto prime = i + i + 3;
+        auto k     = i + prime;
 
-        for (i = 0; i < flags.length; i++)
+        while (k < flags.length)
```
Comment (WalterBright) [inline samples/sieve.d]:
```
foreach (auto k = i + prime; k < flags.length; k += prime)
```

Comment (majiang) [inline samples/sieve.d]: replaced with `iota`, not `for`

File: samples/sieve.d [hunk 3]
```diff
@@ -1,41 +1,28 @@
...
-
-    for (iter = 1;
-         iter <= 10;
-         iter++)
+    immutable max = (1 < args.length)
+        ? args[1].to!size_t
+        : 0x4000;
+    size_t count = 1; // we have 2.
+    // flags[i] = isPrime(2 * i + 3)
+    auto flags = new bool[(max-1)/2];
```
Comment (andralex) [inline samples/sieve.d]: spaces around operators

File: samples/sieve.d [hunk 4]
```diff
@@ -1,41 +1,28 @@
...
+    flags[] = true;
+
+    foreach (i; 0..flags.length)
```
Comment (andralex) [inline samples/sieve.d]: spaces around operators

File: samples/sieve.d [hunk 5]
```diff
@@ -1,41 +1,28 @@
...
     {
-        count   = 0;
-        flags[] = true;
+        if (!flags[i])
+            continue;
+        auto prime = i + i + 3;
+        foreach (k; iota(i+prime, flags.length, prime))
```
Comment (andralex) [inline samples/sieve.d]: spaces around operators

File: samples/sieve.d [hunk 6]
```diff
@@ -1,41 +1,28 @@
...
-                while (k < flags.length)
-                {
-                    flags[k] = false;
-                    k       += prime;
-                }
-
-                count += 1;
-            }
-        }
+        count += 1;
```
Comment (andralex) [inline samples/sieve.d]: `++count;`

**Review by andralex — APPROVED** [review]

File: samples/sieve.d (hunk 3 above)
Comment (majiang) [inline samples/sieve.d]: Fixed

File: samples/sieve.d (hunk 4 above)
Comment (majiang) [inline samples/sieve.d]: Do you mean spaces around `..` ?

File: samples/sieve.d (hunk 5 above)
Comment (majiang) [inline samples/sieve.d]: Fixed

File: samples/sieve.d (hunk 6 above)
Comment (majiang) [inline samples/sieve.d]: Fixed

**Comment by AndrejMitrovic** [comment]
Auto-merge toggled on

**Pull request merged**